pulp_python/app/models.py

```diff
@@ -105,19 +110,23 @@ class PythonImporter(Importer):
     TYPE = 'python'
     projects = models.TextField(null=True)
 
-    def _fetch_inventory(self):
+    def _fetch_inventory(self, current_version):
         """
         Fetch existing contentunits in the repository.
 
+        Args:
+           current_version (pulpcore.plugin.models.RepositoryVersion): the latest pre-existing
+             version or None if one does not exist.
+
         Returns:
             set: of contentunit filenames.
         """
         inventory = set()
-
-        q_set = PythonPackageContent.objects.filter(repositories=self.repository)
-        q_set = q_set.only("filename")
-        for content in (c.cast() for c in q_set):
-            inventory.add(content.filename)
+        if current_version is not None:
```

Comment:
Optional: I'd prefer  to leave this default out of this function. It will be more reusable if it doesn't make assumptions about what to do as if the version isn't specified. I suggest this check should occur as a try/except in the sync code. 

    def sync(self, new_version, base_version):
        try:
            inventory = self._fetch_inventory(base_version)
        except AttributeError: # base_version is None
            inventory = set()